### PR TITLE
{math}[foss/2020b] Dakota v6.15.0 w/ Python 3.8.6

### PR DESCRIPTION
**Pull request by Falkor**
(created using `eb --new-pr`)

As mentioned in comment (see also PR #11072 ), sanity tests requires to use at least 8 cores for MPI processes  (`sbatch --ntasks-per-node 8 [...]`) to successfully complete.

### CODE REVIEW

**Comment by Falkor** [comment]
For some reason, I have trouble to upload build reports, yet the builds were successful on intel broadwell and skylake (CentOS 7) and on AMD epyc (RHEL 8)

File: easybuild/easyconfigs/d/Dakota/Dakota-6.15.0-foss-2020b.eb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+# Disable architecture optimization, otherwise the basic tests fail
+toolchainopts = {'pic': True, 'usempi': True, 'optarch': False}
+
+source_urls = ['https://dakota.sandia.gov/sites/default/files/distributions/public/']
+sources = ["dakota-%(version_major_minor)s.0-public-src-cli.tar.gz"]
+checksums = ['b6d971b7d6fe913aa64fef29cf8fcf55d0e5339f9f55fb164e3cc38716b48f61']
+
+dependencies = [
+    ('Boost',    '1.74.0'),
+    ('OpenBLAS', '0.3.12'),
```
Comment (robert-mijakovic) [inline easybuild/easyconfigs/d/Dakota/Dakota-6.15.0-foss-2020b.eb:27]: OpenBLAS 0.3.12 is already part of foss-2020b. Correct me if I'm wrong, I believe it should be removed.

**Review by robert-mijakovic — CHANGES_REQUESTED** [review]

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
node094.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/120e6ef527474612bef6a0d28610ffd7 for a full test report.

**Comment by robert-mijakovic** [comment]
Test report by @robert-mijakovic
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
mel0385 - Linux Rocky Linux 8.5, x86_64, AMD EPYC 7H12 64-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/ca4a050565a88e501f81327f02494ce6 for a full test report.

File: easybuild/easyconfigs/d/Dakota/Dakota-6.15.0-foss-2020b.eb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ analysis, and quantification of margins and uncertainty with computational
+ models.
+"""
+
+toolchain = {'name': 'foss', 'version': '2020b'}
+# Disable architecture optimization, otherwise the basic tests fail
+toolchainopts = {'pic': True, 'usempi': True, 'optarch': False}
+
+source_urls = ['https://dakota.sandia.gov/sites/default/files/distributions/public/']
+sources = ["dakota-%(version_major_minor)s.0-public-src-cli.tar.gz"]
```
[the remark this answers is not in the document]
Comment (Falkor) [reply easybuild/easyconfigs/d/Dakota/Dakota-6.15.0-foss-2020b.eb]: Done in [f2b2732](https://github.com/easybuilders/easybuild-easyconfigs/pull/15257/commits/f2b273237d21705bf17d369b5da5620506ca384c)

File: easybuild/easyconfigs/d/Dakota/Dakota-6.15.0-foss-2020b.eb (hunk 1 above)
Comment (Falkor) [reply easybuild/easyconfigs/d/Dakota/Dakota-6.15.0-foss-2020b.eb:27]: I don't know but as it is quite a sensitive dependency for Dakota (foss vs intel) I would expect it better to kept explicitly.

Comment (robert-mijakovic) [reply easybuild/easyconfigs/d/Dakota/Dakota-6.15.0-foss-2020b.eb:27]: I tried to install it using foss2021a with OpenBLAS as an explicit dependency and the check for banned linked shared libraries failed.

```
Banned patterns found in linked libraries output for /mnt/tier2/apps/USE/easybuild/staging/2021.1/software/Dakota/6.15.0-foss-2021a/bin/dakota_library_split: '(/|\s)libopenblas\.so'
```

I'm just wondering how you didn't face such an issue?

@jfgrimm @branfosj Can you give me some insights on this?

Comment (robert-mijakovic) [reply easybuild/easyconfigs/d/Dakota/Dakota-6.15.0-foss-2020b.eb:27]: Of course, I can give it myself... the reason is that OpenBLAS was a dependency in foss2020b and the check was not necessary.

In foss2021a, it was replaced by FlexiBLAS and the check is introduced.

File: easybuild/easyconfigs/d/Dakota/Dakota-6.15.0-foss-2020b.eb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+# Disable architecture optimization, otherwise the basic tests fail
+toolchainopts = {'pic': True, 'usempi': True, 'optarch': False}
+
+source_urls = ['https://dakota.sandia.gov/sites/default/files/distributions/public/']
+sources = ["%(namelower)s-%(version_major_minor)s.0-public-src-cli.tar.gz"]
+checksums = ['b6d971b7d6fe913aa64fef29cf8fcf55d0e5339f9f55fb164e3cc38716b48f61']
+
+dependencies = [
+    ('Boost',    '1.74.0'),
+    ('OpenBLAS', '0.3.12'),
```
Comment (Falkor) [inline easybuild/easyconfigs/d/Dakota/Dakota-6.15.0-foss-2020b.eb:27]:
```suggestion
```

**Comment by boegel** [comment]
closing this since `foss/2020b` is deprecated, see https://docs.easybuild.io/policies/toolchains

see also:
* #15883

**Pull request closed without merging**